mobilecoind/src/sync.rs

```diff
@@ -603,6 +615,193 @@ mod test {
+            assert_eq!(utxo.value, test_utils::DEFAULT_PER_RECIPIENT_AMOUNT);
+            assert_eq!(utxo.attempted_spend_height, 0);
+        }
+
+        // Process the last remaining block.
+        let result = sync_monitor(&ledger_db, &mobilecoind_db, &monitor_id, &logger).unwrap();
+        assert_eq!(result, SyncMonitorOk::NoMoreBlocks);
+
+        let monitor_data = mobilecoind_db.get_monitor_data(&monitor_id).unwrap();
+        assert_eq!(monitor_data.next_block, num_blocks as u64);
+
+        let utxos = mobilecoind_db
+            .get_utxos_for_subaddress(&monitor_id, DEFAULT_SUBADDRESS_INDEX)
+            .unwrap();
+        assert_eq!(utxos.len(), num_blocks);
+
+        // Sanity test the utxos.
+        for utxo in utxos {
+            assert!(account0_tx_outs.contains(&utxo.tx_out));
+            assert_eq!(utxo.subaddress_index, 0);
+            assert_eq!(utxo.value, test_utils::DEFAULT_PER_RECIPIENT_AMOUNT);
+            assert_eq!(utxo.attempted_spend_height, 0);
+        }
+
+        // Calling sync_monitor again should not change the results.
+        let result = sync_monitor(&ledger_db, &mobilecoind_db, &monitor_id, &logger).unwrap();
+        assert_eq!(result, SyncMonitorOk::NoMoreBlocks);
+
+        let monitor_data = mobilecoind_db.get_monitor_data(&monitor_id).unwrap();
+        assert_eq!(monitor_data.next_block, num_blocks as u64);
+
+        let utxos = mobilecoind_db
+            .get_utxos_for_subaddress(&monitor_id, DEFAULT_SUBADDRESS_INDEX)
+            .unwrap();
+        assert_eq!(utxos.len(), num_blocks);
+
+        // Sanity test the utxos.
+        for utxo in utxos.iter() {
+            assert!(account0_tx_outs.contains(&utxo.tx_out));
+            assert_eq!(utxo.subaddress_index, 0);
+            assert_eq!(utxo.value, test_utils::DEFAULT_PER_RECIPIENT_AMOUNT);
+            assert_eq!(utxo.attempted_spend_height, 0);
+        }
+
+        // All our utxos should be unique.
+        assert_eq!(HashSet::from_iter(utxos).len(), num_blocks);
+        let receiver = AccountKey::random(&mut rng);
+
+        // Add a block that has a bunch of irrelevant txos.
+        let mut transaction_builder = TransactionBuilder::new(
+            BlockVersion::MAX,
+            Amount::new(Mob::MINIMUM_FEE, Mob::ID),
+            MockFogResolver::default(),
+            EmptyMemoBuilder::default(),
+        )
+        .unwrap();
+        let mut tx_outs = Vec::new();
+        for i in 0..1000 {
+            let TxOutContext { tx_out, .. } = transaction_builder
+                .add_output(Amount::new(10, Mob::ID), &receiver.subaddress(i), &mut rng)
+                .unwrap();
+            tx_outs.push(tx_out);
+        }
+        let start = Instant::now();
+
+        add_txos_to_ledger(&mut ledger_db, BlockVersion::MAX, &tx_outs, &mut rng).unwrap();
+
+        let add_txos_time = start.elapsed();
+        log::info!(logger, "add_txos_to_ledger took {:?}", add_txos_time);
```

Comment:
```suggestion
        log::info!(logger, "add_txos_to_ledger took {add_txos_time:?}");
```
(I'm surprised clippy doesn't catch this)